### PR TITLE
Make a couple of things available for re-use

### PR DESCRIPTION
**Pull request by daviddrysdale**

### CODE REVIEW

**Comment by daviddrysdale** [comment]
Not quite -- there's a future pull request for a different client library that is built on `jsonclient` and also has a need to report rich errors.  I figured moving to a common library is better than copypasta.

**Comment by RJPercival** [comment]
It's a little strange for a package to declare an error type that it doesn't use itself, but provides purely for the convenience of its users.

**Comment by RJPercival** [comment]
It looks like there are a few places in `jsonclient` that could return an `RspError`, rather than just a plain `error` as they are doing at the moment, e.g. https://github.com/google/certificate-transparency-go/blob/9d85f88de63e4b94608dc8750f39a9108e5fe4b6/jsonclient/client.go#L180

**Comment by daviddrysdale** [comment]
Good idea, moved the `RspError` generation down to `jsonclient/` too -- PTAL.

File: client/getentries.go [hunk 1]
```diff
@@ -41,11 +41,8 @@ func (c *LogClient) GetRawEntries(ctx context.Context, start, end int64) (*ct.Ge
 	}
 
 	var resp ct.GetEntriesResponse
-	httpRsp, body, err := c.GetAndParse(ctx, ct.GetEntriesPath, params, &resp)
+	_, _, err := c.GetAndParse(ctx, ct.GetEntriesPath, params, &resp)
 	if err != nil {
```
Comment (RJPercival) [inline client/getentries.go]: nit: could combine lines 44-45.

File: client/logclient.go [hunk 2]
```diff
@@ -210,11 +201,8 @@ func (c *LogClient) GetSTHConsistency(ctx context.Context, first, second uint64)
 		"second": strconv.FormatUint(second, base10),
 	}
 	var resp ct.GetSTHConsistencyResponse
-	httpRsp, body, err := c.GetAndParse(ctx, ct.GetSTHConsistencyPath, params, &resp)
+	_, _, err := c.GetAndParse(ctx, ct.GetSTHConsistencyPath, params, &resp)
 	if err != nil {
```
Comment (RJPercival) [inline client/logclient.go]: nit: could combine lines 213-214.

File: client/logclient.go [hunk 3]
```diff
@@ -229,11 +217,8 @@ func (c *LogClient) GetProofByHash(ctx context.Context, hash []byte, treeSize ui
 		"hash":      b64Hash,
 	}
 	var resp ct.GetProofByHashResponse
-	httpRsp, body, err := c.GetAndParse(ctx, ct.GetProofByHashPath, params, &resp)
+	_, _, err := c.GetAndParse(ctx, ct.GetProofByHashPath, params, &resp)
 	if err != nil {
```
Comment (RJPercival) [inline client/logclient.go]: nit: could combine lines 220-221.

File: client/logclient.go [hunk 4]
```diff
@@ -268,11 +250,8 @@ func (c *LogClient) GetEntryAndProof(ctx context.Context, index, treeSize uint64
 		"tree_size":  strconv.FormatUint(treeSize, base10),
 	}
 	var resp ct.GetEntryAndProofResponse
-	httpRsp, body, err := c.GetAndParse(ctx, ct.GetEntryAndProofPath, params, &resp)
+	_, _, err := c.GetAndParse(ctx, ct.GetEntryAndProofPath, params, &resp)
 	if err != nil {
```
Comment (RJPercival) [inline client/logclient.go]: nit: could combine lines 253-254.

File: jsonclient/client_test.go [hunk 5]
```diff
@@ -225,55 +224,61 @@ func TestGetAndParse(t *testing.T) {
...
+			} else if !strings.Contains(err.Error(), test.wantErr) {
+				t.Errorf("GetAndParse(%q)=_,_,%q; want _, _, error containing %q", test.uri, err.Error(), test.wantErr)
 			}
 			continue
 		}
-		if httpRsp.StatusCode != test.status {
-			t.Errorf("GetAndParse('%s') got status %d; want %d", test.uri, httpRsp.StatusCode, test.status)
+
+		if len(test.wantErr) > 0 {
+			t.Errorf("GetAndParse(%q)=%+v,_,nil; want error matching %q", test.uri, got, test.wantErr)
```
[the remark this answers is not in the document]
Comment (daviddrysdale) [reply jsonclient/client_test.go]: ["when printing structs, the plus flag (%+v) adds field names"](https://golang.org/pkg/fmt/#hdr-Printing)

**Comment by daviddrysdale** [comment]
Left the minor pre-existing things as-is &ndash; I feel I've done enough clean-up in this PR already&hellip;

**Pull request merged**